services.py

```diff
@@ -142,15 +146,31 @@ def get_service(self, id):
             "price_per_unit": service.price_per_unit,
             "unit": service.unit
         })
-    
+    @classmethod
+    def get_services_for_invoice(self, invoice_id):
+        invoice = InvoiceService.get_invoice_by_id(invoice_id)
+        services = []
+        for sr in invoice.service_requests:
+            s = Service.query.get(sr.service_id)
```

Comment:
Never do a query in a loop if you can help it. It's a huge performance dip. You can do a bulk query for everything outside the loop, and build any map you want for a constant lookup time.